core/src/main/java/org/mule/internal/connection/CachedConnectionHandler.java

```diff
@@ -53,24 +55,56 @@ public CachedConnectionHandler(Config config, ConnectionProvider<Config, Connect
     }
 
     /**
-     * On the first invokation to this method, a connection is established using the provided
+     * On the first invocation to this method, a connection is established using the provided
      * {@link #connectionProvider}. That connection is cached and returned.
-     * <p>
-     * Following invokations simply return the same connection.
+     * <p/>
+     * Following invocations simply return the same connection.
      *
      * @return a {@code Connection}
      * @throws ConnectionException   if a {@code Connection} could not be obtained
-     * @throws IllegalStateException if the first invokation is executed while the {@link #muleContext} is stopping or stopped
+     * @throws IllegalStateException if the first invocation is executed while the {@link #muleContext} is stopping or stopped
      */
     @Override
     public Connection getConnection() throws ConnectionException
     {
+        Connection connection = retrieveConnection();
+        ConnectionValidationResult validationResult = connectionProvider.validate(connection);
```

Comment:
shouldn't you call the validateConnection method here instead of doing it directly? Or am I missing something?